core/io/resource_loader.cpp

```diff
@@ -356,8 +358,11 @@ void ResourceLoader::_thread_load_function(void *p_userdata) {
 
 	thread_load_mutex.unlock();
 
-	if (load_nesting == 0 && mq_override) {
-		memdelete(mq_override);
+	if (load_nesting == 0) {
+		if (mq_override) {
+			memdelete(mq_override);
+		}
+		memdelete(load_paths_stack);
```

Comment:
It might be good practice to set pointers to nullptr after calling memdelete to reduce risk of a dangling pointer being inadvertently referenced.

That said, I assume there must already be some sort of guarantee that these variables are not referenced outside of threaded load, so it's not something to change for 4.1 ... just an observation. (I'm not too familiar with this code)

Reply:
You're right. At that point the thread is about to die and no one could access the variable anymore. That said, it is when assumptions fail to be true for whatever reason (code evolution, etc.) when one would have wished to have put guidelines like set-to-null-after-free in practice without further consideration.